src/content.ts

```diff
@@ -464,6 +468,16 @@ function cancelSponsorSchedule(): void {
 function startSponsorSchedule(includeIntersectingSegments: boolean = false, currentTime?: number): void {
     cancelSponsorSchedule();
 
+    // Don't skip if advert playing and reset last checked time
+    if (document.getElementsByClassName('ad-showing').length > 0) {
+        // Reset lastCheckVideoTime
```

Comment:
Why reset this?

Reply:
Without this, if the sponsor is at the start, it won't be skipped after the ad is over, because it thinks that its a duplicate skip, hence why the last checked time is reset